src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -129,18 +128,17 @@ protected sealed override string GenerateFullPathToTool() =>
         protected abstract void AddResponseFileCommands(CommandLineBuilderExtension commandLine);
 
         /// <summary>
-        /// ToolName is only used in cases where <see cref="IsManagedTool"/> returns true.
-        /// It returns the name of the managed assembly, which might not be the path returned by
-        /// GenerateFullPathToTool, which can return the path to e.g. the dotnet executable.
+        /// Thise is the file name the builtin tool that will be executed.
         /// </summary>
         /// <remarks>
-        /// We *cannot* actually call IsManagedTool in the implementation of this method,
-        /// as the implementation of IsManagedTool calls this property. See the comment in
-        /// <see cref="ManagedToolTask.IsManagedTool"/>.
+        /// ToolName is only used in cases where <see cref="UsingBuiltinTool"/> returns true.
```

Comment:
>ToolName [](http://example.com/codeflow?start=12&length=8)

nit: consider renaming to "BuiltinToolName" to reinforce this expectation

Reply:
This cannot be changed because it's a `virtual` defined in `ToolTask`.